sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala

```diff
@@ -419,35 +426,77 @@ abstract class TreeNode[BaseType <: TreeNode[BaseType]] extends Product {
    */
   def nodeName: String = getClass.getSimpleName.replaceAll("Exec$", "")
 
+  /** The argument list to be displayed in tree String. */
+  protected def argumentsForStringOutput(verbose: Boolean): Seq[(String, Any)] = {
```

Comment:
do we really need this?  I understand you are trying to make it very general, but the number of operators that are replacing this seems small.

The ones that want to override this can also just override simpleString.